### PR TITLE
Add `spoc run` command

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This allows to run a seccomp profile for a custom command or binary.

#### Which issue(s) this PR fixes:

Refers to #1482 
#### Does this PR have test?

Yes
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added `spoc run` command for experimental testing of seccomp profiles.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/security-profiles-operator-maintainers PTAL

**Review by JAORMX — APPROVED** [review]

**Comment by JAORMX** [comment]
/lgtm cancel

Let's wait for another maintainer to review.

File: internal/pkg/cli/runner/impl.go
Comment (pjbgf) [inline internal/pkg/cli/runner/impl.go:45]: It would be great if we could have a more generic API, in which we could send different types of security profile enforcer (seccomp, apparmor, selinux, etc..). But as this is internal, we can do that at a later stage.

File: internal/pkg/cli/runner/runner.go
Comment (pjbgf) [inline internal/pkg/cli/runner/runner.go:82]: More for a follow-up: I am not sure the difference between `Init` and `Setup`. For other enforcers, maybe things like `Init` and `Load`, or `Init` and `Enforce`.

Assuming that the former initialises the enforcer and the latter actually applies it to the target application/process.

**Comment by pjbgf** [comment]
Great idea and implementation! Who needs ChatGPT when @saschagrunert is around! :mechanical_arm: :rocket: 

Adding a hold in case other folks may want to comment before the merge.

/lgtm
/hold

**Review by ccojocar — APPROVED** [review]
Great work! 

I am wondering if is not more appropriate to name this command "validate" instead of "run" since this seems more inline with the purpose of the command.

**Comment by saschagrunert** [comment]
Awesome, yeah we can follow-up on the other topics since I'm also planning to push more docs and testing around the features. I just updated the PR to propagate the correct exit code via `spoc run`.

**Comment by saschagrunert** [comment]
/unhold

Thanks for all the reviews!

**Pull request merged**
